tracer/src/Datadog.Trace/SignalFx/Metrics/SignalFxMetricExporter.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Net;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagation;
+using Datadog.Trace.Vendors.ProtoBuf;
+using Datadog.Tracer.SignalFx.Metrics.Protobuf;
+
+namespace Datadog.Trace.SignalFx.Metrics
+{
+    /// <summary>
+    /// Simple exporter that exports metric in SignalFx proto format over http.
+    /// </summary>
+    internal class SignalFxMetricExporter : ISignalFxMetricExporter
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SignalFxMetricExporter));
+        private readonly Uri _metricsEndpointAddress;
+        private readonly string _apiToken;
+        private readonly int _webRequestTimeoutMs;
+
+        public SignalFxMetricExporter(Uri metricsEndpointAddress, string apiToken)
+        {
+            if (metricsEndpointAddress == null)
+            {
+                throw new ArgumentNullException(nameof(metricsEndpointAddress));
+            }
+
+            _metricsEndpointAddress = metricsEndpointAddress;
+            _apiToken = apiToken;
+            _webRequestTimeoutMs = 500;
+        }
+
+        public void Send(DataPointUploadMessage msg)
+        {
+            if (msg == null || msg.datapoints.Count < 1)
```

Comment:
Defensive programming
```suggestion
            if (msg?.datapoints == null || msg.datapoints.Count < 1)
```